File: resources/expected/ros2/datatype-examples/strict/src/datatypes_system_cpp_pkg/src/base_code/producer_producer_base_src.cpp
#include "datatypes_system_cpp_pkg/base_headers/producer_producer_base_src.hpp"

//=================================================
//  D O   N O T   E D I T   T H I S   F I L E
//=================================================

producer_producer_base::producer_producer_base() : Node("producer_producer")
{
    cb_group_ = this->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
    // Setting up connections
    Sys_i_Instance_producer_producer_myBoolean_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Boolean>(
        "Sys_i_Instance_consumer_consumer_myBoolean",
        1);

    Sys_i_Instance_producer_producer_myInteger_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Integer64>(
        "Sys_i_Instance_consumer_consumer_myInteger",
        1);

    Sys_i_Instance_producer_producer_myFloat_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Float64>(
        "Sys_i_Instance_consumer_consumer_myFloat",
        1);

    Sys_i_Instance_producer_producer_myCharacter_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Character>(
        "Sys_i_Instance_consumer_consumer_myCharacter",
        1);

    Sys_i_Instance_producer_producer_myString_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::String>(
        "Sys_i_Instance_consumer_consumer_myString",
        1);

    Sys_i_Instance_producer_producer_myInt8_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Integer8>(
        "Sys_i_Instance_consumer_consumer_myInt8",
        1);

    Sys_i_Instance_producer_producer_myInt16_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Integer16>(
        "Sys_i_Instance_consumer_consumer_myInt16",
        1);

    Sys_i_Instance_producer_producer_myInt32_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Integer32>(
        "Sys_i_Instance_consumer_consumer_myInt32",
        1);

    Sys_i_Instance_producer_producer_myInt64_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Integer64>(
        "Sys_i_Instance_consumer_consumer_myInt64",
        1);

    Sys_i_Instance_producer_producer_myUInt8_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Unsigned8>(
        "Sys_i_Instance_consumer_consumer_myUInt8",
        1);

    Sys_i_Instance_producer_producer_myUInt16_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Unsigned16>(
        "Sys_i_Instance_consumer_consumer_myUInt16",
        1);

    Sys_i_Instance_producer_producer_myUInt32_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Unsigned32>(
        "Sys_i_Instance_consumer_consumer_myUInt32",
        1);

    Sys_i_Instance_producer_producer_myUInt64_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Unsigned64>(
        "Sys_i_Instance_consumer_consumer_myUInt64",
        1);

    Sys_i_Instance_producer_producer_myFloat32_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Float32>(
        "Sys_i_Instance_consumer_consumer_myFloat32",
        1);

    Sys_i_Instance_producer_producer_myFloat64_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::Float64>(
        "Sys_i_Instance_consumer_consumer_myFloat64",
        1);

    Sys_i_Instance_producer_producer_myEnum_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::MyEnum>(
        "Sys_i_Instance_consumer_consumer_myEnum",
        1);

    Sys_i_Instance_producer_producer_myStruct_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::MyStructi>(
        "Sys_i_Instance_consumer_consumer_myStruct",
        1);

    Sys_i_Instance_producer_producer_myArray1_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim>(
        "Sys_i_Instance_consumer_consumer_myArray1",
        1);

    Sys_i_Instance_producer_producer_myArray2_publisher_ = this->create_publisher<datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded>(
        "Sys_i_Instance_consumer_consumer_myArray2",
        1);

    // timeTriggeredCaller callback timer
    periodTimer_ = this->create_wall_timer(std::chrono::milliseconds(1000),
        std::bind(&producer_producer_base::timeTriggeredCaller, this), cb_group_);

    // Used by receiveInputs
    inDataPortTupleVector = {
    };

    // Used by receiveInputs
    inEventPortTupleVector = {
    };

    // Used by sendOutputs
    outPortTupleVector = {
        {&applicationOut_myBoolean, &infrastructureOut_myBoolean, &producer_producer_base::sendOut_myBoolean},
        {&applicationOut_myInteger, &infrastructureOut_myInteger, &producer_producer_base::sendOut_myInteger},
        {&applicationOut_myFloat, &infrastructureOut_myFloat, &producer_producer_base::sendOut_myFloat},
        {&applicationOut_myCharacter, &infrastructureOut_myCharacter, &producer_producer_base::sendOut_myCharacter},
        {&applicationOut_myString, &infrastructureOut_myString, &producer_producer_base::sendOut_myString},
        {&applicationOut_myInt8, &infrastructureOut_myInt8, &producer_producer_base::sendOut_myInt8},
        {&applicationOut_myInt16, &infrastructureOut_myInt16, &producer_producer_base::sendOut_myInt16},
        {&applicationOut_myInt32, &infrastructureOut_myInt32, &producer_producer_base::sendOut_myInt32},
        {&applicationOut_myInt64, &infrastructureOut_myInt64, &producer_producer_base::sendOut_myInt64},
        {&applicationOut_myUInt8, &infrastructureOut_myUInt8, &producer_producer_base::sendOut_myUInt8},
        {&applicationOut_myUInt16, &infrastructureOut_myUInt16, &producer_producer_base::sendOut_myUInt16},
        {&applicationOut_myUInt32, &infrastructureOut_myUInt32, &producer_producer_base::sendOut_myUInt32},
        {&applicationOut_myUInt64, &infrastructureOut_myUInt64, &producer_producer_base::sendOut_myUInt64},
        {&applicationOut_myFloat32, &infrastructureOut_myFloat32, &producer_producer_base::sendOut_myFloat32},
        {&applicationOut_myFloat64, &infrastructureOut_myFloat64, &producer_producer_base::sendOut_myFloat64},
        {&applicationOut_myEnum, &infrastructureOut_myEnum, &producer_producer_base::sendOut_myEnum},
        {&applicationOut_myStruct, &infrastructureOut_myStruct, &producer_producer_base::sendOut_myStruct},
        {&applicationOut_myArray1, &infrastructureOut_myArray1, &producer_producer_base::sendOut_myArray1},
        {&applicationOut_myArray2, &infrastructureOut_myArray2, &producer_producer_base::sendOut_myArray2}
    };

}

//=================================================
//  C o m m u n i c a t i o n
//=================================================

void producer_producer_base::sendOut_myBoolean(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Boolean>(&msg)) {
        Sys_i_Instance_producer_producer_myBoolean_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myBoolean.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myInteger(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Integer64>(&msg)) {
        Sys_i_Instance_producer_producer_myInteger_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myInteger.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myFloat(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Float64>(&msg)) {
        Sys_i_Instance_producer_producer_myFloat_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myFloat.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myCharacter(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Character>(&msg)) {
        Sys_i_Instance_producer_producer_myCharacter_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myCharacter.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myString(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::String>(&msg)) {
        Sys_i_Instance_producer_producer_myString_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myString.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myInt8(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Integer8>(&msg)) {
        Sys_i_Instance_producer_producer_myInt8_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myInt8.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myInt16(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Integer16>(&msg)) {
        Sys_i_Instance_producer_producer_myInt16_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myInt16.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myInt32(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Integer32>(&msg)) {
        Sys_i_Instance_producer_producer_myInt32_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myInt32.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myInt64(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Integer64>(&msg)) {
        Sys_i_Instance_producer_producer_myInt64_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myInt64.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myUInt8(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Unsigned8>(&msg)) {
        Sys_i_Instance_producer_producer_myUInt8_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myUInt8.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myUInt16(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Unsigned16>(&msg)) {
        Sys_i_Instance_producer_producer_myUInt16_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myUInt16.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myUInt32(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Unsigned32>(&msg)) {
        Sys_i_Instance_producer_producer_myUInt32_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myUInt32.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myUInt64(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Unsigned64>(&msg)) {
        Sys_i_Instance_producer_producer_myUInt64_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myUInt64.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myFloat32(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Float32>(&msg)) {
        Sys_i_Instance_producer_producer_myFloat32_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myFloat32.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myFloat64(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::Float64>(&msg)) {
        Sys_i_Instance_producer_producer_myFloat64_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myFloat64.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myEnum(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::MyEnum>(&msg)) {
        Sys_i_Instance_producer_producer_myEnum_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myEnum.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myStruct(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::MyStructi>(&msg)) {
        Sys_i_Instance_producer_producer_myStruct_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myStruct.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myArray1(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim>(&msg)) {
        Sys_i_Instance_producer_producer_myArray1_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myArray1.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::sendOut_myArray2(MsgType msg)
{
    if (auto typedMsg = std::get_if<datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded>(&msg)) {
        Sys_i_Instance_producer_producer_myArray2_publisher_->publish(*typedMsg);
    } else {
        PRINT_ERROR("Sending out wrong type of variable on port myArray2.\nThis shouldn't be possible.  If you are seeing this message, please notify this tool's current maintainer.");
    }
}

void producer_producer_base::put_myBoolean(datatypes_system_cpp_pkg_interfaces::msg::Boolean msg)
{
    enqueue(applicationOut_myBoolean, msg);
}

void producer_producer_base::put_myInteger(datatypes_system_cpp_pkg_interfaces::msg::Integer64 msg)
{
    enqueue(applicationOut_myInteger, msg);
}

void producer_producer_base::put_myFloat(datatypes_system_cpp_pkg_interfaces::msg::Float64 msg)
{
    enqueue(applicationOut_myFloat, msg);
}

void producer_producer_base::put_myCharacter(datatypes_system_cpp_pkg_interfaces::msg::Character msg)
{
    enqueue(applicationOut_myCharacter, msg);
}

void producer_producer_base::put_myString(datatypes_system_cpp_pkg_interfaces::msg::String msg)
{
    enqueue(applicationOut_myString, msg);
}

void producer_producer_base::put_myInt8(datatypes_system_cpp_pkg_interfaces::msg::Integer8 msg)
{
    enqueue(applicationOut_myInt8, msg);
}

void producer_producer_base::put_myInt16(datatypes_system_cpp_pkg_interfaces::msg::Integer16 msg)
{
    enqueue(applicationOut_myInt16, msg);
}

void producer_producer_base::put_myInt32(datatypes_system_cpp_pkg_interfaces::msg::Integer32 msg)
{
    enqueue(applicationOut_myInt32, msg);
}

void producer_producer_base::put_myInt64(datatypes_system_cpp_pkg_interfaces::msg::Integer64 msg)
{
    enqueue(applicationOut_myInt64, msg);
}

void producer_producer_base::put_myUInt8(datatypes_system_cpp_pkg_interfaces::msg::Unsigned8 msg)
{
    enqueue(applicationOut_myUInt8, msg);
}

void producer_producer_base::put_myUInt16(datatypes_system_cpp_pkg_interfaces::msg::Unsigned16 msg)
{
    enqueue(applicationOut_myUInt16, msg);
}

void producer_producer_base::put_myUInt32(datatypes_system_cpp_pkg_interfaces::msg::Unsigned32 msg)
{
    enqueue(applicationOut_myUInt32, msg);
}

void producer_producer_base::put_myUInt64(datatypes_system_cpp_pkg_interfaces::msg::Unsigned64 msg)
{
    enqueue(applicationOut_myUInt64, msg);
}

void producer_producer_base::put_myFloat32(datatypes_system_cpp_pkg_interfaces::msg::Float32 msg)
{
    enqueue(applicationOut_myFloat32, msg);
}

void producer_producer_base::put_myFloat64(datatypes_system_cpp_pkg_interfaces::msg::Float64 msg)
{
    enqueue(applicationOut_myFloat64, msg);
}

void producer_producer_base::put_myEnum(datatypes_system_cpp_pkg_interfaces::msg::MyEnum msg)
{
    enqueue(applicationOut_myEnum, msg);
}

void producer_producer_base::put_myStruct(datatypes_system_cpp_pkg_interfaces::msg::MyStructi msg)
{
    enqueue(applicationOut_myStruct, msg);
}

void producer_producer_base::put_myArray1(datatypes_system_cpp_pkg_interfaces::msg::MyArrayOneDim msg)
{
    enqueue(applicationOut_myArray1, msg);
}

void producer_producer_base::put_myArray2(datatypes_system_cpp_pkg_interfaces::msg::MyArrayUnbounded msg)
{
    enqueue(applicationOut_myArray2, msg);
}

void producer_producer_base::timeTriggeredCaller() {
    receiveInputs();
    timeTriggered();
    sendOutputs();
}

void producer_producer_base::receiveInputs() {
    for (std::tuple<std::queue<MsgType>*, std::queue<MsgType>*> port : inDataPortTupleVector) {
        auto infrastructureQueue = std::get<0>(port);
        if (!infrastructureQueue->empty()) {
            auto msg = infrastructureQueue->front();
            enqueue(*std::get<1>(port), msg);
        }
    }
    for (std::tuple<std::queue<MsgType>*, std::queue<MsgType>*> port : inEventPortTupleVector) {
        auto infrastructureQueue = std::get<0>(port);
        if (!infrastructureQueue->empty()) {
            auto msg = infrastructureQueue->front();
            infrastructureQueue->pop();
            enqueue(*std::get<1>(port), msg);
        }
    }
}

void producer_producer_base::enqueue(std::queue<MsgType>& queue, MsgType val) {
    if (queue.size() >= 1) {
        queue.pop();
    }
    queue.push(val);
}

void producer_producer_base::sendOutputs() {
    for (std::tuple<std::queue<MsgType>*, std::queue<MsgType>*, void (producer_producer_base::*)(MsgType)> port : outPortTupleVector) {
        auto applicationQueue = std::get<0>(port);
        if (applicationQueue->size() != 0) {
            auto msg = applicationQueue->front();
            applicationQueue->pop();
            enqueue(*std::get<1>(port), msg);
        }
    }

    for (std::tuple<std::queue<MsgType>*, std::queue<MsgType>*, void (producer_producer_base::*)(MsgType)> port : outPortTupleVector) {
        auto infrastructureQueue = std::get<1>(port);
        if (infrastructureQueue->size() != 0) {
            auto msg = infrastructureQueue->front();
            infrastructureQueue->pop();
            (this->*std::get<2>(port))(msg);
        }
    }
}

File: resources/expected/ros2/python-datatype-examples/lax/src/datatypes_system_py_pkg/datatypes_system_py_pkg/base_code/producer_producer_base_src.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from queue import Queue
from rclpy.callback_groups import ReentrantCallbackGroup
from datatypes_system_py_pkg_interfaces.msg import Boolean
from datatypes_system_py_pkg_interfaces.msg import Integer64
from datatypes_system_py_pkg_interfaces.msg import Float64
from datatypes_system_py_pkg_interfaces.msg import Character
from datatypes_system_py_pkg_interfaces.msg import String
from datatypes_system_py_pkg_interfaces.msg import Integer8
from datatypes_system_py_pkg_interfaces.msg import Integer16
from datatypes_system_py_pkg_interfaces.msg import Integer32
from datatypes_system_py_pkg_interfaces.msg import Unsigned8
from datatypes_system_py_pkg_interfaces.msg import Unsigned16
from datatypes_system_py_pkg_interfaces.msg import Unsigned32
from datatypes_system_py_pkg_interfaces.msg import Unsigned64
from datatypes_system_py_pkg_interfaces.msg import Float32
from datatypes_system_py_pkg_interfaces.msg import MyEnum
from datatypes_system_py_pkg_interfaces.msg import MyStructi
from datatypes_system_py_pkg_interfaces.msg import MyArrayOneDim
from datatypes_system_py_pkg_interfaces.msg import MyArrayUnbounded
from datatypes_system_py_pkg_interfaces.msg import MyArrayTwoDim

#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================

class producer_producer_base(Node):
    def __init__(self):
        super().__init__("producer_producer")

        self.cb_group_ = ReentrantCallbackGroup()

        # Setting up connections
        self.producer_producer_myBoolean_publisher_ = self.create_publisher(
            Boolean,
            "consumer_consumer_myBoolean",
            1)

        self.producer_producer_myInteger_publisher_ = self.create_publisher(
            Integer64,
            "consumer_consumer_myInteger",
            1)

        self.producer_producer_myFloat_publisher_ = self.create_publisher(
            Float64,
            "consumer_consumer_myFloat",
            1)

        self.producer_producer_myCharacter_publisher_ = self.create_publisher(
            Character,
            "consumer_consumer_myCharacter",
            1)

        self.producer_producer_myString_publisher_ = self.create_publisher(
            String,
            "consumer_consumer_myString",
            1)

        self.producer_producer_myInt8_publisher_ = self.create_publisher(
            Integer8,
            "consumer_consumer_myInt8",
            1)

        self.producer_producer_myInt16_publisher_ = self.create_publisher(
            Integer16,
            "consumer_consumer_myInt16",
            1)

        self.producer_producer_myInt32_publisher_ = self.create_publisher(
            Integer32,
            "consumer_consumer_myInt32",
            1)

        self.producer_producer_myInt64_publisher_ = self.create_publisher(
            Integer64,
            "consumer_consumer_myInt64",
            1)

        self.producer_producer_myUInt8_publisher_ = self.create_publisher(
            Unsigned8,
            "consumer_consumer_myUInt8",
            1)

        self.producer_producer_myUInt16_publisher_ = self.create_publisher(
            Unsigned16,
            "consumer_consumer_myUInt16",
            1)

        self.producer_producer_myUInt32_publisher_ = self.create_publisher(
            Unsigned32,
            "consumer_consumer_myUInt32",
            1)

        self.producer_producer_myUInt64_publisher_ = self.create_publisher(
            Unsigned64,
            "consumer_consumer_myUInt64",
            1)

        self.producer_producer_myFloat32_publisher_ = self.create_publisher(
            Float32,
            "consumer_consumer_myFloat32",
            1)

        self.producer_producer_myFloat64_publisher_ = self.create_publisher(
            Float64,
            "consumer_consumer_myFloat64",
            1)

        self.producer_producer_myEnum_publisher_ = self.create_publisher(
            MyEnum,
            "consumer_consumer_myEnum",
            1)

        self.producer_producer_myStruct_publisher_ = self.create_publisher(
            MyStructi,
            "consumer_consumer_myStruct",
            1)

        self.producer_producer_myArray1_publisher_ = self.create_publisher(
            MyArrayOneDim,
            "consumer_consumer_myArray1",
            1)

        self.producer_producer_myArray2_publisher_ = self.create_publisher(
            MyArrayUnbounded,
            "consumer_consumer_myArray2",
            1)

        self.producer_producer_myArray3_publisher_ = self.create_publisher(
            MyArrayTwoDim,
            "consumer_consumer_myArray3",
            1)

        # timeTriggered callback timer
        self.periodTimer_ = self.create_timer(1000, self.timeTriggered, callback_group=self.cb_group_)

    def timeTriggered(self):
        pass

#=================================================
#  C o m m u n i c a t i o n
#=================================================

    def put_myBoolean(self, msg):
        typedMsg = Boolean()
        typedMsg.data = msg
        self.producer_producer_myBoolean_publisher_.publish(typedMsg)

    def put_myInteger(self, msg):
        typedMsg = Integer64()
        typedMsg.data = msg
        self.producer_producer_myInteger_publisher_.publish(typedMsg)

    def put_myFloat(self, msg):
        typedMsg = Float64()
        typedMsg.data = msg
        self.producer_producer_myFloat_publisher_.publish(typedMsg)

    def put_myCharacter(self, msg):
        typedMsg = Character()
        typedMsg.data = msg
        self.producer_producer_myCharacter_publisher_.publish(typedMsg)

    def put_myString(self, msg):
        typedMsg = String()
        typedMsg.data = msg
        self.producer_producer_myString_publisher_.publish(typedMsg)

    def put_myInt8(self, msg):
        typedMsg = Integer8()
        typedMsg.data = msg
        self.producer_producer_myInt8_publisher_.publish(typedMsg)

    def put_myInt16(self, msg):
        typedMsg = Integer16()
        typedMsg.data = msg
        self.producer_producer_myInt16_publisher_.publish(typedMsg)

    def put_myInt32(self, msg):
        typedMsg = Integer32()
        typedMsg.data = msg
        self.producer_producer_myInt32_publisher_.publish(typedMsg)

    def put_myInt64(self, msg):
        typedMsg = Integer64()
        typedMsg.data = msg
        self.producer_producer_myInt64_publisher_.publish(typedMsg)

    def put_myUInt8(self, msg):
        typedMsg = Unsigned8()
        typedMsg.data = msg
        self.producer_producer_myUInt8_publisher_.publish(typedMsg)

    def put_myUInt16(self, msg):
        typedMsg = Unsigned16()
        typedMsg.data = msg
        self.producer_producer_myUInt16_publisher_.publish(typedMsg)

    def put_myUInt32(self, msg):
        typedMsg = Unsigned32()
        typedMsg.data = msg
        self.producer_producer_myUInt32_publisher_.publish(typedMsg)

    def put_myUInt64(self, msg):
        typedMsg = Unsigned64()
        typedMsg.data = msg
        self.producer_producer_myUInt64_publisher_.publish(typedMsg)

    def put_myFloat32(self, msg):
        typedMsg = Float32()
        typedMsg.data = msg
        self.producer_producer_myFloat32_publisher_.publish(typedMsg)

    def put_myFloat64(self, msg):
        typedMsg = Float64()
        typedMsg.data = msg
        self.producer_producer_myFloat64_publisher_.publish(typedMsg)

    def put_myEnum(self, msg):
        typedMsg = MyEnum()
        typedMsg.data = msg
        self.producer_producer_myEnum_publisher_.publish(typedMsg)

    def put_myStruct(self, msg):
        typedMsg = MyStructi()
        typedMsg.data = msg
        self.producer_producer_myStruct_publisher_.publish(typedMsg)

    def put_myArray1(self, msg):
        typedMsg = MyArrayOneDim()
        typedMsg.data = msg
        self.producer_producer_myArray1_publisher_.publish(typedMsg)

    def put_myArray2(self, msg):
        typedMsg = MyArrayUnbounded()
        typedMsg.data = msg
        self.producer_producer_myArray2_publisher_.publish(typedMsg)

    def put_myArray3(self, msg):
        typedMsg = MyArrayTwoDim()
        typedMsg.data = msg
        self.producer_producer_myArray3_publisher_.publish(typedMsg)

